archunit/src/test/java/com/tngtech/archunit/testutil/assertion/DependenciesAssertion.java

```diff
@@ -112,8 +139,77 @@ public ExpectedDependenciesCreator from(Class<?> origin) {
         }
 
         ExpectedDependencies add(Class<?> origin, Class<?> target) {
-            expectedDependencies.add(ImmutableList.of(origin, target));
+            expectedDependencies.add(new ExpectedDependency(origin, target));
             return this;
         }
+
+        public ExpectedDependencies withDescriptionContaining(String descriptionTemplate, Object... args) {
+            getLast(expectedDependencies).descriptionContaining(descriptionTemplate, args);
+            return this;
+        }
+
+        public ExpectedDependencies inLocation(Class<?> location, int lineNumber) {
+            getLast(expectedDependencies).location(location, lineNumber);
+            return this;
+        }
+    }
+
+    private static class ExpectedDependency {
+        private final Class<?> origin;
+        private final Class<?> target;
+        private Optional<Pattern> descriptionPattern = Optional.absent();
+        private Optional<String> locationPart = Optional.absent();
+
+        ExpectedDependency(Class<?> origin, Class<?> target) {
+            this.origin = origin;
+            this.target = target;
+        }
+
+        boolean matches(Dependency dependency) {
+            if (!dependency.getOriginClass().isEquivalentTo(origin) || !dependency.getTargetClass().isEquivalentTo(target)) {
+                return false;
+            }
+            if (descriptionPattern.isPresent() && !descriptionPattern.get().matcher(dependency.getDescription()).matches()) {
+                return false;
+            }
+            return !locationPart.isPresent() || dependency.getDescription().endsWith(locationPart.get());
+        }
```

Comment:
I would generally agree, but I don't like the necessity to add brackets to support the `||` case :joy: 
But maybe it's still alright, because the two `||` cases are pretty uniformly, so I guess it's still alright to read...
With Java 8 there would certainly be a nicer way to write this :wink: